--- libxenserver/src/Makefile.orig	2015-04-17 19:33:43.000000000 +0200
+++ libxenserver/src/Makefile	2015-10-25 22:02:39.745774106 +0100
@@ -45,9 +45,9 @@
          $(shell curl-config --cflags) \
          -W -Wall -Wmissing-prototypes -Werror -std=c99 $(POS_FLAG)
 
-LDFLAGS = -g $(shell xml2-config --libs) \
-             $(shell curl-config --libs) \
-          -Wl,-rpath,$(shell pwd) $(CYGWIN_LIBXML)
+LDFLAGS = -g
+LIBS = $(shell xml2-config --libs)
+PROG_LIBS = $(LIBS) $(shell curl-config --libs)
 
 # -h for Solaris
 SONAME_LDFLAG ?= -soname
@@ -66,6 +66,10 @@
 
 TARBALL_DEST = libxenserver-2.4
 
+PREFIX = /usr
+INCLUDEDIR = $(PREFIX)/include
+LIBDIR = $(PREFIX)/lib
+
 .PHONY: all
 all: $(TEST_PROGRAMS)
 
@@ -76,28 +80,25 @@
 	ln -sf $< $@
 
 libxenserver.so.2.4: $(LIBXENAPI_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxenserver.so.2 $(SHLIB_CFLAGS) -o $@ $^
+	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxenserver.so.2 $(SHLIB_CFLAGS) -o $@ $^ $(LIBS)
 
 libxenserver.a: $(LIBXENAPI_OBJS)
 	$(AR) rcs libxenserver.a $^
 
 $(TEST_PROGRAMS): test/%: test/%.o libxenserver.so
-	$(CC) -o $@ $< $(TEST_LDFLAGS) $(LDFLAGS)
+	$(CC) -o $@ $< $(TEST_LDFLAGS) $(LDFLAGS) $(PROG_LIBS)
 
 
 .PHONY: install
 install: all libxenserver.a
-	$(INSTALL_DIR) $(DESTDIR)/include/xen/api
-	$(INSTALL_DIR) $(DESTDIR)/lib
-	$(INSTALL_DATA) libxenserver.so.2.4 $(DESTDIR)/lib
-	ln -sf libxenserver.so.2.4 $(DESTDIR)/lib/libxenserver.so.2
-	ln -sf libxenserver.so.2 $(DESTDIR)/lib/libxenserver.so
-	$(INSTALL_DATA) libxenserver.a $(DESTDIR)/lib
-	for i in $(LIBXENAPI_HDRS_INT); do \
-	    $(INSTALL_DATA) $$i $(DESTDIR)/include/; \
-	done
+	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)/xen/api
+	$(INSTALL_DIR) $(DESTDIR)$(LIBDIR)
+	$(INSTALL_DATA) libxenserver.so.2.4 $(DESTDIR)$(LIBDIR)
+	ln -sf libxenserver.so.2.4 $(DESTDIR)$(LIBDIR)/libxenserver.so.2
+	ln -sf libxenserver.so.2 $(DESTDIR)$(LIBDIR)/libxenserver.so
+	$(INSTALL_DATA) libxenserver.a $(DESTDIR)$(LIBDIR)
 	for i in $(LIBXENAPI_HDRS); do \
-	    $(INSTALL_DATA) $$i $(DESTDIR)/include/xen/api; \
+	    $(INSTALL_DATA) $$i $(DESTDIR)$(INCLUDEDIR)/xen/api; \
 	done
 
 
